source/extensions/path/uri_template_lib/uri_template_internal.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "absl/types/variant.h"
+#include "re2/re2.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UriTemplate {
+
+namespace Internal {
+
+/**
+ * String to be concatenated in rewritten path.
+ */
+using Literal = absl::string_view;
+
+/**
+ * Determines what operations to use on the input pattern segment.
+ */
+enum class Operator { PathGlob, TextGlob };
+
+/**
+ * Represents a pattern variable. Variables are included in both path match and rewrite paths.
+ */
+struct Variable {
+  absl::string_view name_;
+
+  // replacement value for the rewrite
+  std::vector<absl::variant<Operator, Literal>> match_;
```

Comment:
Is this the "replacement value", or the pattern which this variable matched? Since one of the options in this vector is "Operate" I would think that it's not actually the replacement value, but maybe I'm wrong?

(In any case capitalize and period please, per the style guide. https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar)

Reply:
I think it's the actual value from looking at parseVariable. I believe the Operation is for wildcards (`*`) while Literal is for strings between `\` .

Reply:
Right. So as you point out, this `Variable` is produced by `parseVariable()` and stored in a `ParsedSegment`. It represents a parsed rewrite pattern, as I understand it. Looking at the unit tests, we see that `parseVariable()` is called with input like`"{v_1=*/abc/**}"` In this case, the `match_` is going to contain 3 elements, [glob, "abc", glob]. These are the elements of the parsed input pattern. I don't think they're they replacement value for the rewrite are they? The rewritten URL will not have "*" or "**" in it right? In other words, doesn't this represent a matched variable in a "match pattern" not a matched variable in a "rewrite pattern"? I think the latter is represented by `ParsedSegment`?

Reply:
I see what you're saying. What if we said something like `holds matched segment from path used for rewrite`

Reply:
I think it contains the matched *pattern* not the matched *path*. So how about:
```
// The pattern which this variable matches.
```

Reply:
changed